src/Raven.Client/Util/StreamWithTimeout.cs

```diff
@@ -2,110 +2,32 @@
 using System.IO;
 using System.Threading;
 using System.Threading.Tasks;
-using Sparrow.Utils;
 
 namespace Raven.Client.Util
 {
-    internal sealed class StreamWithTimeout : Stream, IAsyncDisposable
+    internal sealed class StreamWithTimeout(Stream stream) : Stream
     {
-        private static readonly TimeSpan DefaultWriteTimeout = TimeSpan.FromSeconds(120);
         internal static TimeSpan DefaultReadTimeout { get; } = TimeSpan.FromSeconds(120);
 
-        internal readonly Stream _stream;
+        internal readonly Stream _stream = stream;
         private int _writeTimeout;
         private int _readTimeout;
-        private bool _canBaseStreamTimeoutOnWrite;
-        private bool _canBaseStreamTimeoutOnRead;
         private CancellationTokenSource _writeCts;
         private CancellationTokenSource _readCts;
 
         private long _totalRead = 0;
         private long _totalWritten = 0;
 
-        public StreamWithTimeout(Stream stream)
-        {
-            _stream = stream;
-            SetWriteTimeoutIfNeeded(DefaultWriteTimeout);
```

Comment:
You removed the timeout for all requests.
On `RavenDB.Client` we set the timeout to 12 hours.

- We should still set the timeout for old clients.
- Remove it from the new one.
- Make sure that there is timeout in Kestrel?

Reply:
I'm talking with @ayende about that tomorrow